### PR TITLE
Fix generator leak

### PR DESCRIPTION
**Pull request by florian-huber**
Addresses #24  
Changes to `DataGeneratorAllSpectrums`:
- removed `spectrum_ids`
- added `_exclude_not_selected_inchikeys` method

### CODE REVIEW

File: tests/test_data_generators.py [hunk 1]
```diff
@@ -79,6 +79,76 @@ def test_DataGeneratorAllSpectrums():
...
+                          'JFFHVIUZNPTGGR-WJLGXSQGSA-N',
+                          'JGCSKOVQDXEQHI-UHFFFAOYSA-N',
+                          'VCBNPTWPJQLHQN-NYAJDEOCSA-N',
+                          'ZBAMSLOMNLECFR-IEAZIUSSSA-N']
+    found_inchikeys = test_generator.labels_df.columns.to_list()
+    found_inchikeys.sort()
+    assert found_inchikeys == expected_inchikeys, \
+        "Expected different InChIKeys to remain in labels_df"
+
+    # Test if ibkt exoected labels are returned by generator
```
Comment (svenvanderburg) [inline tests/test_data_generators.py:113]: ibkt exoected?

File: ms2deepscore/data_generators.py [hunk 2]
```diff
@@ -333,6 +335,17 @@ def _exclude_nans_from_labels(labels_df: pd.DataFrame):
             print(f"{n_dropped} nans among {len(labels_df)} labels will be excluded.")
         return clean_df
 
+    def _exclude_not_selected_inchikeys(self):
```
Comment (svenvanderburg) [inline ms2deepscore/data_generators.py:338]: It is not immediately clear from the method name and the docstring from *what* the inchikeys are removed (i.e. labels_df)

**Review by svenvanderburg — COMMENTED** [review]
Nice solution 😄  (I am curious how this affects the results!)

I have left some comments, but I also have a more major suggestion:

If we make `_exclude_not_selected_inchikeys` also exclude the inchikeys also from `binned_spectrums`, I think we can safely leave the `_get_spectrum_with_inchikey` method unaltered in the base class.

Thinking even further it would also be possible to get rid of `spectrum_ids` argument for `DataGeneratorAllSpectrums`, instead one could just pass the list of `binned_spectrums` that one wants to select from!

I have a preference for that last option.

File: tests/test_data_generators.py (hunk 1 above)
Comment (florian-huber) [reply tests/test_data_generators.py:113]: looks like: random keyboard typing while editor still open (multiple screen problem)

**Comment by florian-huber** [comment]
Thanks for the comments @svenvanderburg . I guess I made it unnecessarily complicated 😬 ...
I will remove the `spectrum_ids` which will already solve most of the leaking possibilities.

File: ms2deepscore/data_generators.py (hunk 2 above)
Comment (florian-huber) [reply ms2deepscore/data_generators.py:334]: True. I added `labels_df` as argument and extended the docstring.

File: tests/test_data_generators.py (hunk 1 above)
Comment (florian-huber) [reply tests/test_data_generators.py:108]: Fixed.

**Review by svenvanderburg — APPROVED** [review]
Nice! This really cleans up things :)

**Comment by florian-huber** [comment]
Thanks a lot for reviewing Sven!

**Pull request merged**